File: Previos/Previo 2/herencia.cpp
#include <iostream>
using namespace std;

class Animal { //se define la clase Animal
    public://los miembros son accesibles desde afuera de la clase
        void eat () { //declara la funcion
            cout << "I can eat!" << endl; //imprime un mensaje
        }

        void sleep() { //declara la funcion
            cout << "I can sleep!" << endl;//imprime el mensaje
        }
};

class Dog : public Animal { //se define la clase Dog que hereda publicamente la clase Animal
    public: //los miembros son accesibles desde afuera de la clase
        void bark() { //declara la funcion
            cout << "I can bark! Woof woof!!"  << endl; //imprime mensaje
        }
};

int main() {
    Dog dog1; //crea un objeto de la clase Dog

    dog1.eat(); //llama a la funcion
    dog1.sleep(); //llama a la funcion

    dog1.bark(); //llama a la funcion
    
    return 0;
}

File: Previos/Previo4/herencia.py
#se define la clase
class Animal:
    #atributo y metodo
    name = ""
    #funcion
    def eat(self):
        print("i can eat")
#clase hereda Animal
class Dog(Animal):
    #nuevo metodo en subclase
    def display(self):
        #accede al nombre del atributo usando self
        print("My name is ", self.name)
#crea un objeto de subclase
labrador = Dog()
#accede a atributo de subclase y metodo
labrador.name = "Rohu"
labrador.eat()

#llama a subclase
labrador.display()